python/cudf/cudf/dataframe/categorical.py

```diff
@@ -45,15 +45,114 @@ def codes(self):
         else:
             return Series(data)
 
-    def set_categories(self, new_categories):
+    def to_string(self):
+        if self.ordered:
+            return "[{}]".format(" < ".join(self.categories))
```

Comment:
You should explicitly copy `self.categories` to host here otherwise you will copy from device to host one by one.